core/src/repair/serve_repair.rs

```diff
@@ -510,16 +534,29 @@ impl ServeRepair {
         }
     }
 
+    fn support_coding_repair(cluster_type: ClusterType) -> bool {
+        match cluster_type {
+            ClusterType::Development | ClusterType::Testnet => true,
+            ClusterType::Devnet | ClusterType::MainnetBeta => false,
+        }
+    }
```

Comment:
This does not seem necessary to me.
We should also limit use of `ClusterType` for protocol compatibility changes to exceptional cases.
What is the risk that this is meant to prevent?

Reply:
Since this server side change doesn't yet have a consumer I figured it could be limited to testnet for now. This would reduce backwards compatibility considerations if the API was changed. However, this API is quite simple and mirrors the existing data shred methods so I don't have a problem with removing this check.